frontend/device/input.lua

```diff
@@ -434,6 +435,35 @@ function Input:resetState()
 end
 
 function Input:handleKeyBoardEv(ev)
+    -- Detect loss of contact for the "snow" protocol...
+    if self.snow_protocol then
+        if ev.code == C.BTN_TOUCH and ev.value == 0 then
+            -- Kernel sends it after loss of contact for *all* slots,
+            -- only once the final contact point has been lifted.
+            if #self.MTSlots == 0 then
```

Comment:
Not interested to understand all that, but just reading this, I wonder: why do we handle touch/MT/multitouch slots in a handle**KeyBoard**Ev() function ?!
If this is not an error, it may need a bit of comment saying why :)

Reply:
I think because BTN_TOUCH is a "keyboard" event, contrary to the other touch events.

Reply:
Yep, it's an `EV_KEY` event, while the other handler is for `EV_ABS`... and `EV_SYN` (which includes `SYN_REPORT`, which is where we actually finalize the slot data and send it to GestureManager ;).